ui/src/app/store/bootresource/types/actions.ts

```diff
@@ -0,0 +1,32 @@
+import type { BootResource } from "./base";
+import type { BootResourceSourceType } from "./enum";
+
+export type DeleteImageParams = {
+  id: BootResource["id"];
+};
+
+export type FetchParams = {
+  keyring_data?: string;
+  keyring_filename?: string;
+  source_type: BootResourceSourceType;
+  url: string;
+};
+
+export type SaveOtherParams = {
+  images: string[];
+};
+
+export type SaveUbuntuCoreParams = {
+  images: string[];
+};
+
+export type SaveUbuntuParams = {
+  osystems: UbuntuOsystem[];
+  source_type: BootResourceSourceType;
+};
+
+export type UbuntuOsystem = {
```

Comment:
Should this type be before `SaveUbuntuParams`? It seems to work, but always seems odd when they're defined afterwards, but then it messes up the alphabet!

Reply:
Yeah I'm not sure how to reconcile this. Logically if you were to read top to bottom it makes sense for `SaveUbuntuParams` to be at the top. But if you were looking for something in particular, it's easier to scan things that are alphabetical. But if you were *actually* looking for something, you'd grep it. So I dunno! Maybe it doesn't matter.